### DISCUSSION TITLE
additional_fqdns domains are not added in the server configuration

### DISCUSSION THREAD

**Discussion by shochdoerfer** [php-perfect/ddev-intellij-plugin#181] [Ideas]
Currently, the plugin does not add a server entry for domains configured via the `additional_fqdns` ddev setting. I would assume the same applies to hostnames defined via the `additional_hostnames` setting.

Since our git repository project names and thus our ddev project names are a bit cryptic, we usually define a "user-friendly" version via `additional_fqdns` (also to keep backward compatibility with our old Docker setup). It would be beneficial if all domains defined in `additional_fqdns` and `additional_hostnames` are also properly configured in the server settings. This would allow debugging via xdebug for any of the configured domains.

**Comment by nico-loeber** [comment]
Moin @shochdoerfer,

The additional hostnames and additional_fqdns settings both add to the `urls` entry in the `ddev describe -j` output, so they are genereally available to the plugin to be used for auto configuration.
Currently only the primary url of the project is used to generate a server configuration that is used by xdebug and other tools.
One server can only have one host name in the phpstorm settings, as far as I know.
Could you please describe your exepcted result? Do you configure one server for each of your additional hostnames / fqdns?

**Reply by shochdoerfer** [reply]
> One server can only have one host name in the phpstorm settings, as far as I know.

True, would have been cool if aliases could be configured, but that does not seem to be supported.

Ideally, a server entry for each host name defined in `additional_hostnames` or `additional_fqdns` should be created (basically a copy of the default server since aliasing does not work). For most of our setups, I think we have just one hostname configured. I know we are dealing with multiple hostnames in one specific project, but that's rather the exception than the norm. Here also, it would be cool if I could just use any of the domains for easier debugging without configuring the host configuration myself.

Keeping the different server configs in sync is not needed, I think. That will make things more complicated.

Does this make sense to you?

**Reply by nico-loeber** [reply]
I'm not a fan of the multi server setup with same path mappings as I am not sure how this will affect the internal "run and debug" buttons and stuff like that. Not sure if PHPStorm will pick one random server configuration for that path mapping or what will happen here.

My suggestion is to open a feature request at https://youtrack.jetbrains.com/issues/WI and to ask for server aliases and wait for the feedback. If JetBrains suggests using multiple server configurations we can give it a try to implement it in the plugin.
In case there is a way to add aliases or if they implement that feature, it should be quite simple to add those by the plugin.

> Keeping the different server configs in sync is not needed, I think. That will make things more complicated.

All configurations named "DDEV" are always kept up to date, that's one of the main features of the plugin, so this should be the same for the server settings.
